src/components/paginator-render/paginator-render.tsx

```diff
@@ -0,0 +1,668 @@
+      goToButton: "Go",
+      itemsPerPage: "Items per page:",
+      of: "of ",
+      ellipsis: "...",
+      first: "First",
+      prev: "Prev",
+      next: "Next",
+      last: "Last",
+      unknownPages: "many",
+      unknownItems: "many",
+      showingItems: `Showing ${AMOUNT_ITEMS_IN_PAGE} of ${TOTAL_ITEMS} items`,
+      page: "Page:",
+      showingPage: `Showing ${ACTUAL_PAGE} of ${TOTAL_PAGES} pages`
+    }
+  };
+
+  /**
+   * Event fired when there is a new page selection.
+   */
+  @Event() pageChange: EventEmitter<number>;
+
+  /**
+   * Event fired when the amount of items per page changes.
+   */
+  @Event() itemsPerPageChange: EventEmitter<number>;
+
+  connectedCallback(): void {
+    this.hasHyperlink = Array.isArray(this.model);
+
+    if (this.hasHyperlink) {
+      this.totalPages = (this.model as PaginatorRenderHyperlinkModel).length;
+    } else {
+      this.totalPages =
+        (this.model as PaginatorRenderNumericModel)?.totalPages ?? -1;
+    }
+  }
+
+  #getPaginationRange() {
+    const { actualPage, maxPagesToShowLeft, maxPagesToShowRight } = this;
+    const totalPages = this.totalPages;
+
+    // Set array with pages to show, it starts with the first page on the array.
+    // If page is an ellipsis, array number will be -1
+    const pages = [1]; // Start with the first page
+
+    // Check pages to display at the start and end of the navigation control
+    let startPage = Math.max(actualPage - maxPagesToShowLeft, 2);
+    let endPage = Math.min(actualPage + maxPagesToShowRight, totalPages - 1);
+
+    // If not enough pages to show left, adjust the max pages to show
+    if (actualPage - startPage < maxPagesToShowLeft) {
+      startPage = Math.max(
+        endPage - maxPagesToShowLeft - maxPagesToShowRight,
+        2
+      );
+    }
+
+    // If not enough pages to show right, adjust the max pages to show
+    if (endPage - actualPage < maxPagesToShowRight) {
+      endPage = Math.min(
+        startPage + maxPagesToShowLeft + maxPagesToShowRight,
+        totalPages - 1
+      );
+    }
+
+    // Add pages in range to the array
+    for (let i = startPage; i <= endPage; i++) {
+      pages.push(i);
+    }
+
+    // Validate if ellipsis are needed in range
+    if (totalPages > maxPagesToShowLeft + maxPagesToShowRight) {
+      // Append ellipsis at start if necessary
+      if (startPage > 2) {
+        pages.splice(1, 0, -1); // Append ellipsis after the first page
+      }
+
+      // Append ellipsis at end if necessary
+      if (endPage < totalPages - 1) {
+        pages.push(-1); // Append ellipsis if there are pages after the last visible page
+      }
+    }
+
+    // Append last page at the end if necessary
+    if (totalPages > 1) {
+      pages.push(totalPages); // Ensure the last page is always included
+    }
+
+    return pages;
+  }
+
+  #handlePrev = () => {
+    if (this.actualPage > 1) {
+      this.actualPage -= 1;
+      this.pageChange.emit(this.actualPage);
+    }
+  };
```

Comment:
We can avoid this check by not setting the event if this condition is not met (see next suggestion).

```suggestion
  #handlePrev = () => {
    this.actualPage -= 1;
    this.pageChange.emit(this.actualPage);
  };
```